code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -149,15 +154,20 @@
 
 	//Fill machine with a bag!
 	if(istype(I, /obj/item/storage/bag))
-		var/list/inserted = list()
-		if(SEND_SIGNAL(I, COMSIG_TRY_STORAGE_TAKE_TYPE, /obj/item/reagent_containers/food/snacks/grown, src, limit - length(holdingitems), null, null, user, inserted))
-			for(var/i in inserted)
-				holdingitems[i] = TRUE
-			if(!I.contents.len)
-				to_chat(user, "<span class='notice'>You empty [I] into [src].</span>")
-			else
-				to_chat(user, "<span class='notice'>You fill [src] to the brim.</span>")
-		return TRUE
+		var/obj/item/storage/T = I
+		var/loaded = holdingitems.len
+		for(var/obj/S in T.contents)
+			if(!is_type_in_typecache(S, allowed_grindables))
+				continue
+			if(holdingitems.len >= limit)
+				to_chat(user, "<span class='warning'>\The [src] is full, you can't put anything in!</span>")
+				return TRUE
+			if(SEND_SIGNAL(T, COMSIG_TRY_STORAGE_TAKE, S, src))
+				loaded++
+				holdingitems += S
+		if(loaded)
+			to_chat(user, "<span class='notice'>You insert [loaded] items into \the [src].</span>")
+		return
```

Comment:
A: Why check if the list had items initially? actually why track the list size separately at all? Should it be a counter of how many were added? shouldn't it start at 0 then?
B:does changing the send_signal to the extent that you did effect anything here? tested for feature parity?

Reply:
As to B), feature parity is maintained as far as my testing went.
As to A), whoops